src/ray/gcs/redis_async_context.cc

```diff
@@ -109,6 +118,99 @@ Status RedisAsyncContext::RedisAsyncCommandArgv(redisCallbackFn *fn,
   return Status::OK();
 }
 
-}  // namespace gcs
+void RedisAsyncContext::operate() {
+  if (read_requested_ && !read_in_progress_) {
+    read_in_progress_ = true;
+    socket_.async_read_some(boost::asio::null_buffers(),
```

Comment:
I'm curious about the logic:
- From [doc](https://live.boost.org/doc/libs/1_69_0/doc/html/boost_asio/reference/basic_stream_socket/async_read_some.html), the bytes read is passed to callback
- How do we know all bytes have been read from socket? Same for write logic

Reply:
Good question. This might be due to a historical reason. We can ask other team members why we don't use `async_read` here.

Reply:
cc @jjyao